config/manifest/eksa-components.yaml

```diff
@@ -5983,6 +6002,30 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
+  - packages.eks.amazonaws.com
+  resources:
+  - packagebundlecontrollers
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
```

Comment:
curious about what's driving the need for such broad permissions against the `role` API

Reply:
is there any way to get around adding these permissions to the controller role? It might be convenient to have them here, but this is extremely broad, given it allows the controller to update ANY role (including this role), effectively allowing it to have unbounded permissions

Reply:
@g-gaston asked me to document the reasons for a bunch of these changes, so I'll do that with some code comments, and ping both of you when it's up.

Reply:
ok great thanks Eric!

Reply:
Here's a "more to the point" link: https://github.com/aws/eks-anywhere/blob/99d1c71b48867c6c531cf2085fc47d6921af857f/controllers/cluster_controller_legacy.go#L64

Reply:
I was able to get rid of the update, but I'll have to do a round of testing to be safe.

Reply:
Thanks for this!
Few questions
* We need to create a Role for a particular namespace. Does that mean the namespace name is dynamic?
* The role is for the package controller manager. Does that mean there is one copy of the packages controller (one deployment) running in the management cluster per workload cluster? Or does the new role add permissions to the existing packages controller?
* It seems like it only need to delete rolebindings. If that’s the case, how are the new created roles assigned to anything if it’s not creating new rolebindings?
* For the CRD create operation, couldn’t we create all CRDs when we install packages from the CLI? I feel like at that point we should already know all the CRDs we need.
* What are the secrets that it needs to delete? If it’s not creating any (it doesn’t have permissions to do it), where do those secrets come from?
* For namespaces, the doc mentions it creates a namespace in the workload cluster. Does it also create a namespace in the management cluster? If so, is a dynamically named namespace?

Reply:
> We need to create a Role for a particular namespace. Does that mean the namespace name is dynamic?

Yes, the namespace is of the form "eksa-packages-<workload cluster name>".

> The role is for the package controller manager. Does that mean there is one copy of the packages controller (one deployment) running in the management cluster per workload cluster? Or does the new role add permissions to the existing packages controller?

There is one deployment, regardless of how many clusters. But there is a package bundle controller resource per cluster. So in a situation where you have 1 management and 2 workload clusters, there will be 1 deployment, 1 controller running, but 3 packagebundle controller resources.

> It seems like it only need to delete rolebindings. If that’s the case, how are the new created roles assigned to anything if it’s not creating new rolebindings?

You know, I assumed it already had create permissions from somewhere else? My empirical testing showed that in order to prevent weird errors in the logs I had to be able to remove rolebindings, so I added the permission. It didn't occur to me to track down where the creation permission came from.

> For the CRD create operation, couldn’t we create all CRDs when we install packages from the CLI? I feel like at that point we should already know all the CRDs we need.

Good question. I'll run that by @lewisdiamond as he's got the most knowledge in that area.

> What are the secrets that it needs to delete? If it’s not creating any (it doesn’t have permissions to do it), where do those secrets come from?

There's a k8s cron job installed with the helm chart that keeps registry secrets up to date. They need to be deleted when the cluster is deleted.

> For namespaces, the doc mentions it creates a namespace in the workload cluster. Does it also create a namespace in the management cluster? If so, is a dynamically named namespace?

Yes and yes.

In the management cluster, it creates eksa-packages-<cluster name>. In the workload cluster, it creates "eksa-packages".

Reply:
I found a regression that was causing the CRDs to be installed, even though they already were. Fixed that, and now we don't need CRDs permissions at all. Yay.

I also looked more closely at the Roles being created. The key role being created is the `&lt;workload cluster name&gt;-eks-anywhere-packages-manager-role`, which is created within the workload-cluster-specific namespace `eksa-packages-&lt;workload cluster name&gt;` So that's a dynamically named role in a dynamically create namespace, hence the create and matching delete permissions, and since we don't know the namespace name ahead of time, we can't restrict them to particular namespaces.

The code's comment is updated to reflect these changes.

Reply:
After resetting the permissions model in my head, plus a lot of chats with @g-gaston and @lewisdiamond, I stripped things down to brass tacks, and was able to remove nearly all the permissions. The only one that remains is full control over package bundle controller resources, which signal the packages controller that it needs to install things or remove them or whatever. So that seems totally reasonable to me. No new roles. :)

The namespace of the package bundle controller resources could be further restricted with a namespace (eksa-pacakges), but with the way kubebuilder works, that means we'd be required to define a matching role named "eksa-packages", and I don't think that buys us anything, since no one but us should be modifying package bundle controller resources anyway, as we're the ones that created the CRD in the first place. :)